lib/modules/datasource/github-releases/cache/cache-base.ts

```diff
@@ -365,4 +377,25 @@ export abstract class AbstractGithubDatasourceCache<
 
     return result;
   }
+
+  public isChangelogReleaseFresh(
+    changelogRelease: ChangeLogRelease | undefined,
+    now: DateTime,
+    updateDuration: DurationLike,
+    cacheItems: Record<string, StoredItem>
+  ): boolean {
+    if (changelogRelease) {
+      const changelogReleaseTime = DateTime.fromISO(
+        changelogRelease.date.toString()
+      );
+      if (
+        !cacheItems[changelogRelease.version] &&
+        changelogReleaseTime.plus(updateDuration) >= now
```

Comment:
This needs commenting or moving into functions or something like that so that it's easier to understand in future

Reply:
Did something in-between by extracting each condition into self-describing variable